File: library/src/main/java/com/chaos/aol/external/ArtNative.java
package com.chaos.aol.external;

import android.content.Context;

import com.chaos.aol.info.FieldData;

import java.util.List;

import me.weishu.reflection.Reflection;

public class ArtNative {

    static {
        System.loadLibrary("aol-art");
    }

    public static void init(Context context) {
        Reflection.unseal(context);
        nativeInit();
    }

    public static native void nativeInit();

    public static native List<FieldData> nativeGetFields(Class<?> clz, boolean isStatic);

    public static native int getClassSize(Class<?> clz);
}

File: library/src/main/java/com/chaos/aol/external/ArtNative.kt
package com.chaos.aol.external

import android.content.Context
import com.chaos.aol.info.JField
import me.weishu.reflection.Reflection

object ArtNative {
    fun init(context: Context?) {
        Reflection.unseal(context)
        nativeInit()
    }

    fun getStaticFields(clz: Class<*>): List<JField> {
        return nativeGetFields(clz, true)
    }

    fun getInstanceFields(clz: Class<*>): MutableList<JField> {
        return nativeGetFields(clz, false)
    }

    private external fun nativeInit()
    private external fun nativeGetFields(clz: Class<*>, isStatic: Boolean): MutableList<JField>
    external fun getClassSize(clz: Class<*>): Int

    init {
        System.loadLibrary("aol-art")
    }
}